spec/operators/shareReplay-spec.ts

```diff
@@ -0,0 +1,167 @@
+    const expected1   =     '-1-2-3----4-#';
+    const subscriber2 = hot('    b|       ').mergeMapTo(shared);
+    const expected2   =     '    23----4-#';
+    const subscriber3 = hot('        c|   ').mergeMapTo(shared);
+    const expected3   =     '        3-4-#';
+
+    expectObservable(subscriber1).toBe(expected1);
+    expectObservable(subscriber2).toBe(expected2);
+    expectObservable(subscriber3).toBe(expected3);
+    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  });
+
+  it('should multicast an empty source', () => {
+    const source = cold('|');
+    const sourceSubs =  '(^!)';
+    const shared = source.shareReplay(1);
+    const expected =    '|';
+
+    expectObservable(shared).toBe(expected);
+    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  });
+
+  it('should multicast a never source', () => {
+    const source = cold('-');
+    const sourceSubs =  '^';
+
+    const shared = source.shareReplay(1);
+    const expected =    '-';
+
+    expectObservable(shared).toBe(expected);
+    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  });
+
+  it('should multicast a throw source', () => {
+    const source = cold('#');
+    const sourceSubs =  '(^!)';
+    const shared = source.shareReplay(1);
+    const expected =    '#';
+
+    expectObservable(shared).toBe(expected);
+    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  });
+
+  it('should replay results to subsequent subscriptions if source completes, bufferSize=2', () => {
+    const source =     cold('-1-2-----3-|        ');
+    const shared = source.shareReplay(2);
+    const sourceSubs =      '^          !        ';
+    const subscriber1 = hot('a|                  ').mergeMapTo(shared);
+    const expected1   =     '-1-2-----3-|        ';
+    const subscriber2 = hot('    b|              ').mergeMapTo(shared);
+    const expected2   =     '    (12)-3-|        ';
+    const subscriber3 = hot('               (c|) ').mergeMapTo(shared);
+    const expected3   =     '               (23|)';
+
+    expectObservable(subscriber1).toBe(expected1);
+    expectObservable(subscriber2).toBe(expected2);
+    expectObservable(subscriber3).toBe(expected3);
+    expectSubscriptions(source.subscriptions).toBe(sourceSubs);
+  });
+
+  it('should completely restart for subsequent subscriptions if source errors, bufferSize=2', () => {
+    const source =     cold('-1-2-----3-#               ');
+    const shared = source.shareReplay(2);
+    const sourceSubs1 =     '^          !               ';
+    const subscriber1 = hot('a|                         ').mergeMapTo(shared);
+    const expected1   =     '-1-2-----3-#               ';
+    const subscriber2 = hot('    b|                     ').mergeMapTo(shared);
+    const expected2   =     '    (12)-3-#               ';
+    const subscriber3 = hot('               (c|)        ').mergeMapTo(shared);
+    const expected3   =     '               -1-2-----3-#';
```

Comment:
Ignore me. It's correct. (because the source errored, it didn't *complete*, so we create a new ReplaySubject for the next multicast)